contracts/account/modules/ERC7579Validator.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.27;
+
+import {ERC7579Module} from "./ERC7579Module.sol";
+import {IERC7579Validator, MODULE_TYPE_VALIDATOR} from "@openzeppelin/contracts/interfaces/draft-IERC7579.sol";
+import {PackedUserOperation} from "@openzeppelin/contracts/interfaces/draft-IERC4337.sol";
+import {ERC4337Utils} from "@openzeppelin/contracts/account/utils/draft-ERC4337Utils.sol";
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+/**
+ * @dev Abstract validator module for ERC-7579 accounts.
+ *
+ * This contract provides the base implementation for signature validation in ERC-7579 accounts.
+ * Developers must implement the onInstall, onUninstall, and {_isValidSignatureWithSender} function in derived contracts to
+ * define the specific signature validation logic.
+ *
+ * Example usage:
+ *
+ * ```solidity
+ * contract MyValidatorModule is ERC7579Validator {
+ *     function onInstall(bytes calldata data) public override {
+ *         // Install logic here
+ *     }
```

Comment:
We should not recommend overriding without calling super, someone doing this will skip the `ModuleInstalledReceived` and `ModuleUninstalledReceived` events